src/atom.rs

```diff
@@ -99,6 +99,31 @@ impl<Static> Atom<Static> {
         }
     }
 
+    /// For the atom!() macros
+    #[inline(always)]
+    #[doc(hidden)]
+    pub const fn pack_inline(n: u64, len: u8) -> Self {
+        // Reverse byte order if big-endian
+        let n = if cfg!(target_endian = "big") {
+            ((n & 0x000000000000FF00) << 48)
+                | ((n & 0x0000000000FF0000) << 32)
+                | ((n & 0x00000000FF000000) << 16)
+                | (n & 0x000000FF00000000)
+                | ((n & 0x0000FF0000000000) >> 16)
+                | ((n & 0x00FF000000000000) >> 32)
+                | ((n & 0xFF00000000000000) >> 48)
+        } else {
+            n
+        };
```

Comment:
I think [this](https://doc.rust-lang.org/nightly/core/primitive.u64.html#method.to_le) is equivalent?
```suggestion
        let n = n.to_le();
```

Reply:
Actually no! That's not quite right. It needs to reverse the top 7 bytes, not the whole `u64`. I've pushed a commit which clarifies this.